File: 1441.用栈操作构建数组.cpp
/*
 * @lc app=leetcode.cn id=1441 lang=cpp
 *
 * [1441] 用栈操作构建数组
 */
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> ans;
        int num = 0;
        for (int i: target) {
            ans.push_back("Push");
            for (int j = 0; j < i - num - 1; j++) {
                ans.push_back("Pop");
                ans.push_back("Push");
            }
            num = i;
        }
        return ans;
    }
};
// @lc code=end

File: 1441.用栈操作构建数组.py
#
# @lc app=leetcode.cn id=1441 lang=python3
#
# [1441] 用栈操作构建数组
#

# @lc code=start
from typing import List


class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        arr = []
        num = 0
        for i in target:
            arr.append('Push')
            for j in range(i - num - 1):
                arr.append('Pop')
                arr.append('Push')
            num = i
        return arr
# @lc code=end

